### PR TITLE
Fix airflow_db_cleanup script for Airflow 2

### LINKED ISSUE #7835: airflow_db_cleanup fails on TaskInstance and TaskReschedule tables in Composer 2.x
## In which file did you encounter the issue?

composer/workflows/airflow_db_cleanup.py

### Did you change the file? If so, how?

No changes to the file, other than making the default for airflow_db_cleanup__max_db_entry_age_in_days 90 days.  

## Describe the issue

When allowing the DAG to execute per the schedule, all of the tables are processed except for TaskInstance and TaskReschedule which produce the following error:

[2022-04-26, 14:51:42 UTC] {airflow_db_cleanup_v1.py:292} INFO - Running Cleanup Process...
[2022-04-26, 14:51:42 UTC] {taskinstance.py:1702} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/python3.8/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1330, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/opt/python3.8/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1457, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/opt/python3.8/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1513, in _execute_task
    result = execute_callable(context=context)
  File "/opt/python3.8/lib/python3.8/site-packages/airflow/operators/python.py", line 174, in execute
    return_value = self.execute_callable()
  File "/opt/python3.8/lib/python3.8/site-packages/airflow/operators/python.py", line 185, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/airflow/gcs/dags/airflow_db_cleanup_v1.py", line 295, in cleanup_function
    query = session.query(airflow_db_model).options(
  File "/opt/python3.8/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 1619, in options
    return self._options(False, *args)
  File "<string>", line 2, in _options
  File "/opt/python3.8/lib/python3.8/site-packages/sqlalchemy/orm/base.py", line 227, in generate
    fn(self, *args[1:], **kw)
  File "/opt/python3.8/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 1638, in _options
    opt.process_query(self)
  File "/opt/python3.8/lib/python3.8/site-packages/sqlalchemy/orm/strategy_options.py", line 176, in process_query
    self._process(query, True)
  File "/opt/python3.8/lib/python3.8/site-packages/sqlalchemy/orm/strategy_options.py", line 677, in _process
    val._bind_loader(
  File "/opt/python3.8/lib/python3.8/site-packages/sqlalchemy/orm/strategy_options.py", line 829, in _bind_loader
    raise sa_exc.ArgumentError(
sqlalchemy.exc.ArgumentError: mapper option expects string key or list of attributes

Googling for the `sqlalchemy.exc.ArgumentError: mapper option expects string key or list of attributes` didn't help me, but I admit I'm not versed in SQLAlchemy, so I'll try this issue.

When we were using Airflow 1.x/Composer 1.x this version of the script worked:  [airflow_db_cleanup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/composer/airflow_1_samples/airflow_db_cleanup.py)

### PR DESCRIPTION
**Pull request by khokharnikunj8**
Fixed column name in TaskSchedule and TaskInstance in Airflow workflow

## Description

Fixes  b/230505340 
Fixes #7835 

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Comment by leahecole** [comment]
LGTM 

[Airflow run in my env](https://screenshot.googleplex.com/7nmHwL4sLsrDveT)

**Review by leahecole — APPROVED** [review]

**Comment by khokharnikunj8** [comment]
Thanks @mhenc for pointing out the err. I've updated the code. Please take a look. Thanks!

**Comment by khokharnikunj8** [comment]
@parthea the comments from #7835 are addressed too. Thanks!

**Review by leahecole — APPROVED** [review]
Nik showed me two environments where it's running successfully, LGTM.

File: composer/workflows/airflow_db_cleanup.py [hunk 1]
```diff
@@ -279,6 +279,11 @@ def print_configuration_function(**context):
 
 def cleanup_function(**context):
 
+    if AIRFLOW_VERSION > ['2', '2', '5']:
+        logging.error("The script is not supported for this Airflow Version. "
+                      "Skipped deleting the data.")
+        return
+
```
Comment (leahecole) [inline composer/workflows/airflow_db_cleanup.py:286]:
```suggestion

```

**Review by parthea — APPROVED** [review]
Added one minor observation, otherwise LGTM

File: composer/workflows/airflow_db_cleanup.py [hunk 2]
```diff
@@ -81,7 +81,7 @@
 ALERT_EMAIL_ADDRESSES = []
 # Length to retain the log files if not already provided in the conf. If this
 # is set to 30, the job will remove those files that arE 30 days old or older.
-
+AIRFLOW_VERSION = os.getenv("MAJOR_VERSION", "2.2.0").split(".")
```
[the remark this answers is not in the document]
Comment (khokharnikunj8) [reply composer/workflows/airflow_db_cleanup.py:84]: Initially I used the similar approach. But some python checks were failing in github. So, I changed it to get some default version. WDYT?

[the remark this answers is not in the document]
Comment (parthea) [reply composer/workflows/airflow_db_cleanup.py:84]: Sounds good

File: composer/workflows/airflow_db_cleanup.py [hunk 3]
```diff
@@ -279,6 +279,10 @@ def print_configuration_function(**context):
 
 def cleanup_function(**context):
 
+    if AIRFLOW_VERSION > ['2', '2', '5']:
```
Comment (ben-marengo-msmg) [inline composer/workflows/airflow_db_cleanup.py:282]: hi all - is this a "better safe than sorry" guard? or is there genuinely something happening in >2.2.5 that will render this whole script useless?

**Pull request merged**